### PR TITLE
[SPARK-40010][PYTHON][DOCS] Make pyspark.sql.window examples self-contained

### PR DESCRIPTION
**Pull request by dcoliversun**
### What changes were proposed in this pull request?

This PR proposes to improve the examples in `pyspark.sql.window` by making each example self-contained with a brief explanation and a bit more realistic example.

### Why are the changes needed?

To make the documentation more readable and able to copy and paste directly in PySpark shell.

### Does this PR introduce _any_ user-facing change?

Yes, it changes the documentation

### How was this patch tested?

Manually ran each doctest.

### CODE REVIEW

File: python/pyspark/sql/window.py [hunk 1]
```diff
@@ -196,11 +208,23 @@ def rangeBetween(start: int, end: int) -> "WindowSpec":
         --------
         >>> from pyspark.sql import Window
         >>> from pyspark.sql import functions as func
-        >>> from pyspark.sql import SQLContext
-        >>> sc = SparkContext.getOrCreate()
-        >>> sqlContext = SQLContext(sc)
-        >>> tup = [(1, "a"), (1, "a"), (2, "a"), (1, "b"), (2, "b"), (3, "b")]
-        >>> df = sqlContext.createDataFrame(tup, ["id", "category"])
+        >>> df = spark.createDataFrame(
```
Comment (HyukjinKwon) [inline python/pyspark/sql/window.py:211]: you will probably have to define `spark` in the bottom.

**Comment by HyukjinKwon** [comment]
Looks pretty good otherwise

File: python/pyspark/sql/window.py (hunk 1 above)
Comment (dcoliversun) [reply python/pyspark/sql/window.py:217]: Yeah. without define `spark`, pyspark tests failed. Updated in https://github.com/apache/spark/pull/37450/commits/34623307de7809c76cbcfe864d2a3b4df98a61c5

**Review by HyukjinKwon — APPROVED** [review]

**Review by Yikun — APPROVED** [review]
LGTM, and next, you might want to add parameters description for `orderBy` in a separate PR.

https://github.com/apache/spark/blob/663b2087af9bba5d9026b6c98548ce9075235e5a/python/pyspark/sql/window.py#L85

**Comment by dcoliversun** [comment]
Thanks all @HyukjinKwon @Yikun   
I'd like to create a sub-task of [SPARK-40005](https://issues.apache.org/jira/browse/SPARK-40005) to add parameters description for `pyspark.sql.window`, not only `orderBy` method.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**